x/fees/types/msg.go

```diff
@@ -50,10 +55,12 @@ func (msg MsgRegisterFee) ValidateBasic() error {
 		return sdkerrors.Wrapf(err, "invalid contract address %s", msg.ContractAddress)
 	}
 
-	// WithdrawAddress can be omitted and it will default to DeployerAddress
-	if msg.WithdrawAddress != "" {
-		if _, err := sdk.AccAddressFromBech32(msg.WithdrawAddress); err != nil {
-			return sdkerrors.Wrapf(err, "invalid withdraw address address %s", msg.WithdrawAddress)
+	if msg.WithdrawerAddress != "" {
+		if _, err := sdk.AccAddressFromBech32(msg.WithdrawerAddress); err != nil {
+			return sdkerrors.Wrapf(err, "invalid withdraw address %s", msg.WithdrawerAddress)
+		}
+		if msg.DeployerAddress == msg.WithdrawerAddress {
+			return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "please specify an empty withdraw address instead of setting the deployer address")
```

Comment:
I don't think `msg.DeployerAddress == msg.WithdrawerAddress` should raise an error.

From a user perspective it seems unintuitive to get an error for redundant, but correct information. 
An empty `msg.WithdrawerAddress` will set `withdrawerAddress` to `msg.DeployerAddress` after all.

Reply:
@MalteHerrmann True! I'll remove it here as this will be handled in `msg_server.go` (like you mentioned).